src/parser/core/Events.js

```diff
@@ -78,11 +78,18 @@ const Events = {
   get absorbed() {
     return new EventFilter('absorbed');
   },
-  /**
+   /**
+   * This event is called when a beacon transfer occurs.
+   * @returns {EventFilter}}
+   */
+  get beacontransfer() {
```

Comment:
Please don’t add spec specific event types to the Events object. Instead make it a static prop on the class fabricating the event. I don’t remember what other analyzer has that already, but there are examples in the codebase

Reply:
I finally had a moment to look for the example, see 

https://github.com/WoWAnalyzer/WoWAnalyzer/blob/9100029083007fb7cfaa1b5375f58fb777f1714b/src/parser/paladin/holy/modules/beacons/BeaconHealSource.js#L32-L34

Reply:
Took me a moment to realize that actually the event you want, so it's already there, you just got to use it. heh.

Reply:
I am importing BeHealSource
`import BeaconHealSource from '../../../holy/modules/beacons/BeaconHealingDone.js';`
Adding the dependencies:
```
  static dependencies = {
    beaconHealSource: BeaconHealSource,
  };
```
and adding a listener for the event:
`this.addEventListener(this.beaconHealSource.beacontransfer.by(SELECTED_PLAYER), this.onBeaconTransfer);`
But I am getting a type error when I compile attempt to add the listener.
Unhandled Rejection (TypeError): Cannot read property 'by' of undefined

I think I am using it is the same was as the example @MartijnHols  gave.
Do I need to add another new EventFilter for beacontransfer?

Reply:
AFAIK, `BeaconHealSource` doesn't need to be added as a dependency (if you only use it for the event filter). Import the class, don't add as a dependency, and then use this instead:

```javascript
this.addEventListener(BeaconHealSource.beacontransfer.by(SELECTED_PLAYER), this.onBeaconTransfer);
```

`beacontransfer` is a static getter, bound to the class, not an instance of it.

Reply:
I got rid of the dependency and changed to listen for: `BeaconHealSource.beacontransfer.by(SELECTED_PLAYER)`
But I am still getting the same error where it does not like the `.by(SELECTED_PLAYER)` method.

Reply:
`this.beaconHealSource.beacontransfer.by(SELECTED_PLAYER)` should work since it's a non-static class property. It is used by BeaconHealingDone too.

Reply:
Thanks, I'll give it a go.  Hopefully I can get this PR finished tonight!